backend/src/main/kotlin/code/nebula/cipherquest/repository/QuestionRepository.kt

```diff
@@ -0,0 +1,68 @@
+package code.nebula.cipherquest.repository
+
+import code.nebula.cipherquest.models.requests.QuestionRequest
+import code.nebula.cipherquest.repository.entities.Question
+import org.springframework.ai.document.Document
+import org.springframework.ai.vectorstore.VectorStore
+import org.springframework.data.repository.NoRepositoryBean
+import org.springframework.jdbc.core.JdbcTemplate
+import java.sql.ResultSet
+
+@NoRepositoryBean
+abstract class QuestionRepository<S : QuestionRequest, T : Question>(
+    private val vectorStore: VectorStore,
+    private val jdbcTemplate: JdbcTemplate,
+    private val tableName: String,
+    private val additionalColumns: Map<String, (S) -> Any> = emptyMap(),
```

Comment:
`additionalMetadata` could be a better name for the field because we are managing metadata and not columns.